lib/file-store.js

```diff
@@ -43,7 +42,39 @@ module.exports.upload = async (display_filename, contents, type, assessment_inst
         instance_question_id,
         user_id,
         authn_user_id,
+        storage_type,
     ];
+
+    // uploads to ec2 filesystem
+    if (storage_type === this.storageTypes.FileSystem) {
+        debug('upload()');
+        // Make a filename to store the file. We use a UUIDv4 as the filename,
+        // and put it in two levels of directories corresponding to the first-3
+        // and second-3 characters of the filename.
+
+        if (config.filesRoot == null) throw new Error('config.filesRoot is null, which does not allow uploads');
+
+        debug(`upload() : mkdir ${dir}`);
+        await fsPromises.mkdir(dir, {recursive: true, mode: 0o700});
+        debug(`upload(): writeFile ${filename}`);
+        await fsPromises.writeFile(filename, buffer, {mode: 0o600});
+    }
+
+    // uploads to s3 bucket
+    if (storage_type === this.storageTypes.S3) {
```

Comment:
Can we make this an `else if` from the previous `if (storage_type === this.storageTypes.FileSystem)` and add a final `else` clause that throws an exception if we have an unknown storage type?

Reply:
Most certainly.